frame/election-provider-multi-phase/test-staking-e2e/src/lib.rs

```diff
@@ -53,12 +53,14 @@ fn log_current_time() {
 
 #[test]
 fn block_progression_works() {
-	ExtBuilder::default().build_and_execute(|| {
+	let (mut ext, pool_state, _) = ExtBuilder::default().build_offchainify();
```

Comment:
It would be nicer if we could just do something like
`ExtBuilder::default().add_ocw().execute_with(...)`

Can't we get the pool_state from externalities?

Reply:
The problem is that we need to pass the `pool_state` within the closure in `ext.execute_with`, which doesn't have access to `ext`'s state anymore. the state is only useful during the construction of the test externalities.